### PR TITLE
[Do not merge] [Cross Train] QnA cross-train core functionality

### PR DESCRIPTION
**Pull request by feich-ms**
1. Support QnA cross-train core function bases on lu cross-train. 
2. Add back test cases to cover lu and qna cross training
3. Rename luCrossTrainer.js to crossTrainer.js and expose cross-train.js to handle files input and output
4. Add cli file cross-train.ts in command folder for test purpose. Will remove the file once review done.
5. Pull fathers's qna questions to children's interruption intents
6. Empty qna file will not pull corresponding lu utterances as qna question
7. config file of mapping rules should be specified by flags.config parameter. If not, it will read default config.json from the folder specified by --in
8. Remove duplicated utterances in interruption intent to make sure it has no duplications with other intent utterances in the lu file
9. Move cross train CLI to new created package/cross-train and core crossTrainer.js and cross-train.js to packages/lu/src/parser/cross-train. All the detail tests for cross train core functionalities are in packages/lu/test/parser/cross-train. CLI tests are in package/cross-train/test folder.

### CODE REVIEW

**Comment by feich-ms** [comment]
@vishwacsena @munozemilio could you please set up a review meeting based on your convenient time.

**Comment by vishwacsena** [comment]
@feich-ms any update on this?

**Comment by feich-ms** [comment]
@vishwacsena, I was working on some new requirements about lubuild these days such as exposing functions for Composer and CLI refinements. I'm picking up the cross train work today. Will give some updates tomorrow. Sorry for the delay.

**Comment by feich-ms** [comment]
@vishwacsena, the updates based on our last sync meeting for qna cross training are pushed. You can checkout this branch and test it by CLI.

**Comment by feich-ms** [comment]
@munozemilio could you help to give some review suggestions?

**Comment by feich-ms** [comment]
@vishwacsena, I have updated the PR to resolve all the feedbacks from you and Dong above, and also added test cases. You can try again when you have time. Thanks.

File: packages/cross-train/src/commands/cross-train.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {Command, flags} from '@microsoft/bf-cli-command'
+const crossTrain = require('./../../node_modules/@microsoft/bf-lu/lib/parser/cross-train/cross-train')
+const path = require('path')
+
+export default class CrossTrian extends Command {
```
Comment (boydc2014) [inline packages/cross-train/src/commands/cross-train.ts]: CrossTrain

File: packages/cross-train/src/commands/cross-train.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {Command, flags} from '@microsoft/bf-cli-command'
+const crossTrain = require('./../../node_modules/@microsoft/bf-lu/lib/parser/cross-train/cross-train')
```
Comment (boydc2014) [inline packages/cross-train/src/commands/cross-train.ts]: I don't know why we have to hard code path here?

File: packages/cross-train/src/commands/cross-train.ts (hunk 1 above)
Comment (feich-ms) [reply packages/cross-train/src/commands/cross-train.ts]: fixed

File: packages/cross-train/src/commands/cross-train.ts (hunk 2 above)
Comment (feich-ms) [reply packages/cross-train/src/commands/cross-train.ts]: fixed. Can replace with require('@microsoft/bf-lu/lib/parser/cross-train/cross-train')

**Comment by feich-ms** [comment]
@vishwacsena, the cross train CLI has been moved to packages/cross-train folder. Please try it in packages/cross-train/bin with command: run cross-train --in '../test/fixtures/testcases/interuption' --config './mapping_rules.json'.

**Comment by boydc2014** [comment]
A few new minor issues
1. when config is an invalid json, the message is unclear

> C:\Users\donglei\code\botframework-cli\packages\cross-train\bin>run.cmd cross-train --in c:\Users\donglei\cttests\tests1
> Unknown error during execution. Please file an issue on https://github.com/microsoft/botframework-cli/issues
> undefined

2. when the result folder is not there, not creating the folder for me (don't have to fix)

> C:\Users\donglei\code\botframework-cli\packages\cross-train\bin>run.cmd cross-train --in=c:\Users\donglei\cttests\tests1--out=c:\Users\donglei\cttests\tests1\result
> Unknown error during execution. Please file an issue on https://github.com/microsoft/botframework-cli/issues
> ENOENT: no such file or directory, stat 'c:\Users\donglei\cttests\tests1--out=c:\Users\donglei\cttests\tests1\result'

3. de-dup seems not always consistent, 

Still see  dup utterance in the result like following, but another result don't have this, not sure what happened 
> # _Interuption
> - trigger b
> - trigger b

**Comment by feich-ms** [comment]
Close this PR since the core cross-train library without CLI is picked up in PR #601.

**Pull request closed without merging**